src/core/transaction.zig

```diff
@@ -472,15 +475,15 @@ pub const CompiledKeys = struct {
             } else unreachable;
         }
 
-        std.debug.assert(account_keys.items.len == account_keys_buf.len);
+        std.debug.assert(account_keys.items.len == num_account_keys);
 
         const header = MessageHeader{
             .num_required_signatures = @intCast(readonly_signers_end),
             .num_readonly_signed_accounts = @intCast(readonly_signers_end - writable_signers_end),
             .num_readonly_unsigned_accounts = @intCast(account_keys.items.len - writable_non_signers_end),
         };
 
-        return .{ header, account_keys_buf };
+        return .{ header, try account_keys.toOwnedSlice(allocator) };
```

Comment:
`toOwnedSlice` is only an extra allocation if it has to reallocate in order to shrink the allocation (ie the capacity != items.len and `resize` fails). Because it's initialized with `initCapacity`, the `toOwnedSlice` should effectively be a noop.

Though personally I would concur this is not immediately obvious by itself, I prefer the previous code wherein we allocate a fixed-length slice and then use it to back a `std.ArrayListUnmanaged` which we expect to fill.